src/Mailer/Message.php

```diff
@@ -1747,12 +1758,19 @@ protected function decodeForHeader(string $text): string
     /**
      * Read the file contents and return a base64 version of the file contents.
      *
-     * @param string $path The absolute path to the file to read.
+     * @param string|\Psr\Http\Message\UploadedFileInterface $file The absolute path to the file to read
+     *   or UploadedFileInterface instance.
      * @return string File contents in base64 encoding
      */
-    protected function readFile(string $path): string
+    protected function readFile($file): string
     {
-        return chunk_split(base64_encode((string)file_get_contents($path)));
+        if (is_string($file)) {
+            $content = (string)file_get_contents($file);
+        } else {
+            $content = (string)$file->getStream();
```

Comment:
Shouldn't this be the type we check for instead of `string`?

Reply:
Umm.. no?  `$file` can either be a string filesystem path or an `UploadedFileInterface` instance, that's what the `is_string()` check is for.

Reply:
But we could also get an `array` or other invalid value which will turn into a bad method call here. Should we give an exception on a non-string/non uploadinterface object?

Reply:
I added check and exception in `setAttachments()` where various other checks are done too.